### PR TITLE
Add support for the Bright benchmark 

### PR DESCRIPTION
**Pull request by swj0419**
## Checklist for adding MMTEB dataset 
Reason for dataset addition: (Bright dataset, https://huggingface.co/datasets/xlangai/BRIGHT)
 
* [x]  I have tested that the dataset runs with the `mteb` package.
* [x]  I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  
    * [x]  `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
    * [x]  `intfloat/multilingual-e5-small`
 * [x]  I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
 * [x]  I have considered the size of the dataset and reduced it if it is too big 
 * [x]  Run tests locally to make sure nothing is broken using `make test`. (NOTE: make test didn't work for me with error (pytest: error: unrecognized arguments: -n), but `pytest --durations=5` passes)
 * [x]  Run the formatter to format the code using `make lint`.
 * [ ]  I have added points for my submission to the [POINTS.md](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) file. (i will add these information when the arxiv version of our pape is out)

### CODE REVIEW

**Comment by xiamengzhou** [comment]
Thanks @swj0419 !

I saw many benchmarks have separate task files for each subtask, e.g., CQA in [this page](https://github.com/embeddings-benchmark/mteb/tree/main/mteb/tasks/Retrieval/eng). Conceptually, do you think we should maintain one BRIGHT page or multiple ones?

Also, would you suggest that we separate standard and long settings to two files, given that the main_score is different? We use ndcg_at_10 for the standard setting, and recall_at_1 for the long setting.

File: mteb/abstasks/TaskMetadata.py [hunk 1]
```diff
@@ -235,6 +235,23 @@ def _check_eval_langs(cls, eval_langs):
...
+            "robotics",
+            "stackoverflow",
+            "sustainable_living",
+            "pony",
+            "leetcode",
+            "aops",
+            "theoremqa_theorems",
+            "theoremqa_questions",
+        ]:
+            return
```
Comment (KennethEnevoldsen) [inline mteb/abstasks/TaskMetadata.py:253]:
```suggestion
```

File: mteb/tasks/Retrieval/eng/BrightRetrieval.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    "stackoverflow",
+    "sustainable_living",
+    "pony",
+    # "leetcode",
+    # "aops",
+    # "theoremqa_theorems",
+    # "theoremqa_questions"
+]
+
+DOMAINS_langs = {domain: [domain] for domain in DOMAINS}
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Retrieval/eng/BrightRetrieval.py:24]:
```suggestion
DOMAINS_langs = {domain: [LANG_CODES_FOR_THE_SPLIT] for split in DOMAINS}
```
The format should be like this

File: mteb/tasks/Retrieval/eng/BrightRetrieval.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        main_score="ndcg_at_10",
+        date=None,
+        form=None,
+        domains=None,
+        task_subtypes=["Article retrieval"],
+        license=None,
+        socioeconomic_status=None,
+        annotations_creators=None,
+        dialect=[],
+        text_creation=None,
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Retrieval/eng/BrightRetrieval.py:53]: These needs to be filled out. Especially domain is probably a good case here.

File: mteb/tasks/Retrieval/eng/BrightRetrieval.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        bibtex_citation="""
+            @misc{BRIGHT,
+            title={BRIGHT: A Realistic and Challenging Benchmark for Reasoning-Intensive Retrieval},
+            author={Su, Hongjin and Yen, Howard and Xia, Mengzhou and Shi, Weijia and Muennighoff, Niklas and Wang, Han-yu and Liu, Haisu and Shi, Quan and Siegel, Zachary S and Tang, Michael and Sun, Ruoxi and Yoon, Jinsung and Arik, Sercan O and Chen, Danqi and Yu, Tao},
+            year={2024},
+            }
+        """,
+        n_samples=None,
+        avg_character_length=None,
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Retrieval/eng/BrightRetrieval.py:62]: all required. You can use the calc metric method on the retrieval task

File: mteb/tasks/Retrieval/eng/BrightRetrieval.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+EVAL_SPLITS = ["standard", "long"]
+
+
+class BrightRetrieval(MultilingualTask, AbsTaskRetrieval):
+    metadata = TaskMetadata(
+        name="BrightRetrieval",
+        dataset={
+            "path": "xlangai/BRIGHT",
+            "revision": "a75a0eb",
+            # "name": "narrativeqa",
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Retrieval/eng/BrightRetrieval.py:36]:
```suggestion
```

File: mteb/tasks/Retrieval/eng/BrightRetrieval.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    "economics",
+    "psychology",
+    "robotics",
+    "stackoverflow",
+    "sustainable_living",
+    "pony",
+    # "leetcode",
+    # "aops",
+    # "theoremqa_theorems",
+    # "theoremqa_questions"
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Retrieval/eng/BrightRetrieval.py:21]:
```suggestion
    "leetcode",
    "aops",
    "theoremqa_theorems",
    "theoremqa_questions"
```

**Review by KennethEnevoldsen — COMMENTED** [review]
Thanks for this PR!

Looks like we are also missing scores as well. Will you submit them within this PR.

File: mteb/tasks/Retrieval/eng/BrightRetrieval.py (hunk 3 above)
Comment (xiamengzhou) [reply mteb/tasks/Retrieval/eng/BrightRetrieval.py:53]:
```suggestion
        date=("2024-03-01", "2024-06-01"),
        form="written",
        domains=["Non-fiction"],
        task_subtypes=["Article retrieval"],
        license="CC-BY-4.0",
        socioeconomic_status="low",
        annotations_creators="derived",
        dialect=[],
        text_creation="found",
```

**Review by hongjin-su — CHANGES_REQUESTED** [review]
Add a running script

**Review by swj0419 — COMMENTED** [review]
Thanks Hongjin

**Comment by KennethEnevoldsen** [comment]
Hi @swj0419 I will just ping you here as it seems like there is still some changes lacking

**Comment by xiamengzhou** [comment]
@KennethEnevoldsen @swj0419 Added missing info! Anything else we should check?

**Comment by KennethEnevoldsen** [comment]
Thanks @xiamengzhou!

We will need to update from the main branch (TaskMetadata had an update in #1054) and we will need to add in the descriptive statistics. Otherwise I believe it is ready

File: mteb/tasks/Retrieval/eng/BrightRetrieval.py [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+        },
+        reference="https://huggingface.co/datasets/xlangai/BRIGHT",
+        description=("Bright retrieval dataset."),
+        type="Retrieval",
+        category="s2p",
+        eval_splits=EVAL_SPLITS,
+        eval_langs=DOMAINS_langs,
+        main_score="ndcg_at_10",
+        date=("2024-03-01", "2024-06-01"),
+        form="written",
```
Comment (Muennighoff) [inline mteb/tasks/Retrieval/eng/BrightRetrieval.py:45]:
```suggestion
        form=["written"],
```

File: mteb/abstasks/TaskMetadata.py [hunk 8]
```diff
@@ -233,6 +233,7 @@ def _check_eval_langs(cls, eval_langs):
 
     @staticmethod
     def _check_language_code(code):
+
```
Comment (Muennighoff) [inline mteb/abstasks/TaskMetadata.py:236]:
```suggestion
```

**Review by Muennighoff — APPROVED** [review]
Fixed the outstanding things I think; Also up here now: https://huggingface.co/spaces/mteb/leaderboard?task=retrieval&language=bright

Great work on this PR! ❤️

**Pull request merged**